internal/bminventory/inventory.go

```diff
@@ -2240,6 +2240,9 @@ func (b *bareMetalInventory) getLogFileForDownload(ctx context.Context, clusterI
 		}
 		downloadFileName = fmt.Sprintf("%s_%s_%s.tar.gz", sanitize.Name(c.Name), role, sanitize.Name(hostutil.GetHostnameForMsg(hostObject)))
 	case string(models.LogsTypeController):
+		if c.Cluster.ControllerLogsCollectedAt == strfmt.DateTime(time.Time{}) {
+			return "", "", common.NewApiError(http.StatusNotFound, errors.Errorf("Controller Logs for cluster %s were not found", clusterId))
```

Comment:
This should return 409, not 404.  404 would suggest that the cluster doesn't exist,  I know you probably copie the behavior of host logs - would appreciate if you updated that as well to 409.

Reply:
done.